approvals/approvals/api.py

```diff
@@ -151,10 +169,10 @@ def set_status_to_approved(doc: Document, method: str | None = None, automatic=F
 
 
 @frappe.whitelist()
-def reject_document(doc: Document, role=None, comment: str = "", method: str | None = None):
-	doc = frappe._dict(json.loads(doc)) if isinstance(doc, str) else doc
-	doc = frappe.get_doc(doc.doctype, doc.name)
+def reject_document(doc: Document | str, role=None, comment: str = "", method: str | None = None):
+	doc = frappe.get_doc(json.loads(doc)) if isinstance(doc, str) else doc
 	doc.save(ignore_permissions=True)
+	# TODO: should we check if this method exists before calling?
 	doc.set_status(update=True, status="Rejected")
```

Comment:
@agritheory The `set_status` call isn't present in the base document classes, which means that this _may_ potentially fail in the future. If we're confident about usage though, then I think we can keep the status quo here.

Reply:
@Alchez No, we can't assume this will always be the case. Where a workflow exists, we can infer a rejected state. Where it does not, we should take no action. I think we should use the `apply_workflow` function.

Reply:
Do we want to infer a rejected state based on some keywords? Or use a custom field?